plugins/event_source/gcp_pubsub.py

```diff
@@ -0,0 +1,68 @@
+"""
+gcp_pubsub.py
+
+An ansible-events event source module for receiving events from GCP Pub/Sub
+
+Arguments:
+    project_id:      The GCP project name
+    subscription_id: The name of the topic to pull messages from
+    max_messages:    The number of messages to retreive
+                     Default 3
+    retry_deadline:  How long to keep retrying in seconds
+                     Default 300
+
+Example:
+
+    - ansible.eda.gpc_pubsub:
+        project_id: "{{ project_id }}"
+        subscription_id: "{{ subscription_id }}"
+        max_messages: "{{ max_messages }}"
+        retry_deadline: "{{ retry_deadline }}"
+
+"""
+
+import asyncio
+from typing import Any, Dict
+
+from google.api_core import retry
+from google.cloud import pubsub_v1
```

Comment:
Does pubsub_v1 support asyncio? Typically pull-and-sleep is not the best approach.

Reply:
As far as I see this plugin would block the python thread.[ It could implement futures + callbacks](https://cloud.google.com/pubsub/docs/samples/pubsub-subscriber-async-pull#pubsub_subscriber_async_pull-python) or, I think easier, [use an asyncio library compatible](https://github.com/talkiq/gcloud-aio/tree/master/pubsub)